include/cglm/struct/mat3.h

```diff
@@ -38,12 +38,12 @@
 #include "../mat3.h"
 #include "vec3.h"
 
-#define GLMS_MAT3_IDENTITY_INIT  {1.0f, 0.0f, 0.0f,                          \
-                                  0.0f, 1.0f, 0.0f,                          \
-                                  0.0f, 0.0f, 1.0f}
-#define GLMS_MAT3_ZERO_INIT      {0.0f, 0.0f, 0.0f,                          \
-                                  0.0f, 0.0f, 0.0f,                          \
-                                  0.0f, 0.0f, 0.0f}
+#define GLMS_MAT3_IDENTITY_INIT  {{1.0f, 0.0f, 0.0f,                         \
+                                   0.0f, 1.0f, 0.0f,                         \
+                                   0.0f, 0.0f, 1.0f}}
```

Comment:
Maybe we could use matrix initializers like this instead:

```C
#define GLMS_MAT3_IDENTITY_INIT  {{1.0f, 0.0f, 0.0f},                         \
                                  {0.0f, 1.0f, 0.0f},                         \
                                  {0.0f, 0.0f, 1.0f}}
```

I have checked the array api and they are initialized like this. Maybe we could use the same

Reply:
No, that doesn't work. It picks the wrong field to initialize. The only thing that works is a designated initializer:

```c
#define GLMS_MAT3_IDENTITY_INIT  {.col = {{1.0f, 0.0f, 0.0f},                \
                                          {0.0f, 1.0f, 0.0f},                \
                                          {0.0f, 0.0f, 1.0f}}}
```

But that's C99-only and I think MSVC doesn't support it.

Reply:
> No, that doesn't work. It picks the wrong field to initialize.

May I ask why wrong? We are initializing the whole struct at once. 

> But that's C99-only and I think MSVC doesn't support it.

Agree. Let's not use designated initializers.

Reply:
I think I figured it out. It was picking the anonymous struct over `col` because that's what comes first. Just re-ordering the `union` so that `col` comes first fixes it. I'll amend this PR.